---
+++ Makefile
@@ -70,7 +70,7 @@ ui-lint:
 assets: ui-install ui-build
 
 .PHONY: assets-compress
-assets-compress: assets
+assets-compress: # assets
 	@echo '>> compressing assets'
 	scripts/compress_assets.sh
 
@@ -139,7 +139,7 @@ plugins/plugins.go: plugins.yml plugins/generate.go
 plugins: plugins/plugins.go
 
 .PHONY: build
-build: assets npm_licenses assets-compress plugins common-build
+build: assets-compress plugins common-build
 
 .PHONY: bench_tsdb
 bench_tsdb: $(PROMU)
